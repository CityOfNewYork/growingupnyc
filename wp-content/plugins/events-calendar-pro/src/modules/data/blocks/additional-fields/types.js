/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_PRO_STORE } from '@moderntribe/events-pro/data/prefix';

export const SET_ADDITIONAL_FIELD_INITIAL_STATE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_INITIAL_STATE`;
export const ADD_ADDITIONAL_FIELD = `${ PREFIX_EVENTS_PRO_STORE }/ADD_ADDITIONAL_FIELD`;
export const REMOVE_ADDITIONAL_FIELD = `${ PREFIX_EVENTS_PRO_STORE }/REMOVE_ADDITIONAL_FIELD`;

export const SET_ADDITIONAL_FIELD_LABEL = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_LABEL`;
export const SET_ADDITIONAL_FIELD_IS_PRISTINE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_IS_PRISTINE`;
export const SET_ADDITIONAL_FIELD_VALUE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_VALUE`;
export const SET_ADDITIONAL_FIELD_TYPE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_TYPE`;
export const SET_ADDITIONAL_FIELD_OPTIONS = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_OPTIONS`;
export const SET_ADDITIONAL_FIELD_DIVIDER_LIST = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_DIVIDER_LIST`;
export const SET_ADDITIONAL_FIELD_DIVIDER_END = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_DIVIDER_END`;
export const APPEND_ADDITIONAL_FIELD_VALUE = `${ PREFIX_EVENTS_PRO_STORE }/APPEND_ADDITIONAL_FIELD_VALUE`;
export const REMOVE_ADDITIONAL_FIELD_VALUE = `${ PREFIX_EVENTS_PRO_STORE }/REMOVE_ADDITIONAL_FIELD_VALUE`;
export const SET_ADDITIONAL_FIELD_CHANGE = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_CHANGE`;
export const SET_ADDITIONAL_FIELD_META_KEY = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_META_KEY`;
export const SET_ADDITIONAL_FIELD_OUTPUT = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_OUTPUT`;
export const SET_ADDITIONAL_FIELD_BLUR = `${ PREFIX_EVENTS_PRO_STORE }/SET_ADDITIONAL_FIELD_BLUR`;
