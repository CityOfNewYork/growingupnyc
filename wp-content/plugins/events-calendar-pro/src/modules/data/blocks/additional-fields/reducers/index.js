/**
 * Internal dependencies
 */
import fields from './fields';

export default fields;
