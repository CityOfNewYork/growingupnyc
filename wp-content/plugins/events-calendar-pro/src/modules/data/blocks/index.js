/**
 * Internal dependencies
 */
import * as constants from './constants';
import * as recurring from './recurring';
import reducer from './reducer';

export default reducer;
export { constants };
export { recurring };
