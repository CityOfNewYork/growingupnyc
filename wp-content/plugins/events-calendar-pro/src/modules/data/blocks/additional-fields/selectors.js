/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import identity from 'lodash/identity';
import includes from 'lodash/includes';
import uniq from 'lodash/uniq';

/**
 * Internal dependencies
 */
import { constants } from '@moderntribe/common/data/plugins';

export const getPlugin = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ];
export const getBlocks = createSelector(
	[ getPlugin ],
	( plugin ) => plugin.blocks,
);

export const getAdditionalFields = createSelector(
	[ getBlocks ],
	( blocks ) => blocks.additionalFields,
);

export const getFieldName = ( state, props ) => props.name;

export const getIds = createSelector(
	[ getAdditionalFields ],
	( fields ) => fields.allIds,
);

export const getFieldAsObjects = createSelector(
	[ getAdditionalFields ],
	( fields ) => fields.byId,
);

export const getFieldBlock = createSelector(
	[ getFieldAsObjects, getFieldName ],
	( fields, name ) => fields[ name ] || {},
);

export const getFieldDividerList = createSelector(
	[ getFieldBlock ],
	( field ) => field.dividerList,
);

export const getFieldDividerEnd = createSelector(
	[ getFieldBlock ],
	( field ) => field.dividerEnd,
);

export const getFieldOutput = createSelector(
	[ getFieldBlock ],
	( field ) => field.output,
);

export const getFieldMetaKey = createSelector(
	[ getFieldBlock ],
	( field ) => field.metaKey,
);

export const getFieldType = createSelector(
	[ getFieldBlock ],
	( field ) => field.type,
);

export const getFieldLabel = createSelector(
	[ getFieldBlock ],
	( field ) => field.label || '',
);

export const getFieldValue = createSelector(
	[ getFieldBlock ],
	( field ) => field.value,
);

export const getTextFieldValue = createSelector(
	[ getFieldValue ],
	( value ) => value || '',
);

export const getTextAreaOutput = createSelector(
	[ getFieldOutput ],
	( output = '' ) => {
		return output.split( '\n' ).filter( identity );
	},
);

export const getFieldIsPristine = createSelector(
	[ getFieldBlock ],
	( field ) => field.isPristine,
);

export const getFieldOptions = createSelector(
	[ getFieldBlock ],
	( field ) => field.options || [],
);

export const getFieldOptionsWithLabels = createSelector(
	[ getFieldOptions ],
	( options ) => {
		return options.map( ( option ) => ( { value: option, label: option } ) );
	},
);

export const getFieldDropdownValue = createSelector(
	[ getFieldBlock ],
	( field ) => ( { value: field.value, label: field.value } ),
);

export const getFieldDropdownOutput = createSelector(
	[ getFieldBlock ],
	( field ) => field.output,
);

export const getFieldCheckboxValue = createSelector(
	[ getTextFieldValue ],
	( value ) => uniq( value.split( '|' ) ),
);

export const getFieldCheckboxOptions = createSelector(
	[ getFieldCheckboxValue, getFieldOptionsWithLabels ],
	( values, optionsWithLabels ) => {
		return optionsWithLabels.map( ( option ) => ( {
			...option,
			isChecked: includes( values, option.value ),
		} ) );
	},
);
