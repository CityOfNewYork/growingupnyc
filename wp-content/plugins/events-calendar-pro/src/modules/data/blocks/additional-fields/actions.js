/**
 * Internal dependencies
 */
import { types } from '@moderntribe/events-pro/data/blocks/additional-fields';
import { SET_ADDITIONAL_FIELD_OUTPUT } from './types';

export const setInitialState = ( props ) => ( {
	type: types.SET_ADDITIONAL_FIELD_INITIAL_STATE,
	payload: props,
} );

export const setFieldChange = ( name ) => ( {
	type: types.SET_ADDITIONAL_FIELD_CHANGE,
	payload: {
		name,
	},
} );

export const setFieldBlur = ( name ) => ( {
	type: types.SET_ADDITIONAL_FIELD_BLUR,
	payload: {
		name,
	},
} );

export const setFieldBlurWithType = ( name, type ) => ( {
	type: `${ types.SET_ADDITIONAL_FIELD_BLUR }/${ type }`,
	payload: {
		name,
	},
} );

export const addField = ( name ) => ( {
	type: types.ADD_ADDITIONAL_FIELD,
	payload: {
		name,
	},
} );

export const removeField = ( name ) => ( {
	type: types.REMOVE_ADDITIONAL_FIELD,
	payload: {
		name,
	},
} );

export const setFieldLabel = ( name, label ) => ( {
	type: types.SET_ADDITIONAL_FIELD_LABEL,
	payload: {
		name,
		label,
	},
} );

export const setFieldIsPristine = ( name, isPristine ) => ( {
	type: types.SET_ADDITIONAL_FIELD_IS_PRISTINE,
	payload: {
		name,
		isPristine,
	},
} );

export const setFieldValue = ( name, value ) => ( {
	type: types.SET_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const appendFieldValue = ( name, value ) => ( {
	type: types.APPEND_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const removeFieldValue = ( name, value ) => ( {
	type: types.REMOVE_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const setFieldType = ( name, type ) => ( {
	type: types.SET_ADDITIONAL_FIELD_TYPE,
	payload: {
		name,
		type,
	},
} );

export const setFieldOptions = ( name, options ) => ( {
	type: types.SET_ADDITIONAL_FIELD_OPTIONS,
	payload: {
		name,
		options,
	},
} );

export const setFieldDividerList = ( name, dividerList ) => ( {
	type: types.SET_ADDITIONAL_FIELD_DIVIDER_LIST,
	payload: {
		name,
		dividerList,
	},
} );

export const setFieldDividerEnd = ( name, dividerEnd ) => ( {
	type: types.SET_ADDITIONAL_FIELD_DIVIDER_END,
	payload: {
		name,
		dividerEnd,
	},
} );

export const setFieldMetaKey = ( name, metaKey ) => ( {
	type: types.SET_ADDITIONAL_FIELD_META_KEY,
	payload: {
		name,
		metaKey,
	},
} );

export const setFieldOutput = ( name, output ) => ( {
	type: SET_ADDITIONAL_FIELD_OUTPUT,
	payload: {
		name,
		output,
	},
} );
