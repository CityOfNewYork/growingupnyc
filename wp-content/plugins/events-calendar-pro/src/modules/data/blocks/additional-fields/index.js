/**
 * Internal dependencies
 */
import reducer from './reducers';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import sagas from './sagas';

export default reducer;
export { types, actions, selectors, sagas };
