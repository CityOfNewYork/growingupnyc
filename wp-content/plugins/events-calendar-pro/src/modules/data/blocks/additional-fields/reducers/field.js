/**
 * Internal dependencies
 */
import { types } from '@moderntribe/events-pro/data/blocks/additional-fields';
import { __ } from '@wordpress/i18n';

export const DEFAULT_STATE = {
	isPristine: true,
	value: '',
	type: '',
	options: [],
	dividerList: ', ',
	dividerEnd: __( ' and ', 'tribe-events-calendar-pro' ),
	label: '',
	metaKey: '',
	output: '',
};

export default ( state = DEFAULT_STATE, action ) => {
	const { payload = {} } = action;
	switch ( action.type ) {
		case types.SET_ADDITIONAL_FIELD_LABEL:
			return {
				...state,
				label: payload.label,
			};
		case types.SET_ADDITIONAL_FIELD_VALUE:
			return {
				...state,
				value: payload.value,
			};
		case types.SET_ADDITIONAL_FIELD_TYPE:
			return {
				...state,
				type: payload.type,
			};
		case types.SET_ADDITIONAL_FIELD_IS_PRISTINE:
			return {
				...state,
				isPristine: payload.isPristine,
			};
		case types.SET_ADDITIONAL_FIELD_OPTIONS:
			return {
				...state,
				options: payload.options,
			};
		case types.SET_ADDITIONAL_FIELD_DIVIDER_LIST:
			return {
				...state,
				dividerList: payload.dividerList,
			};
		case types.SET_ADDITIONAL_FIELD_DIVIDER_END:
			return {
				...state,
				dividerEnd: payload.dividerEnd,
			};
		case types.SET_ADDITIONAL_FIELD_META_KEY:
			return {
				...state,
				metaKey: payload.metaKey,
			}
		case types.SET_ADDITIONAL_FIELD_OUTPUT:
			return {
				...state,
				output: payload.output,
			}
		default:
			return state;
	}
}
