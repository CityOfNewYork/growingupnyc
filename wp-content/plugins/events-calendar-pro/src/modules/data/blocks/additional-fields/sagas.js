/**
 * External dependencies
 */
import { takeEvery, put, select, all } from 'redux-saga/effects';
import noop from 'lodash/noop';
import isEmpty from 'lodash/isEmpty';
import identity from 'lodash/identity';

/**
 * Internal dependencies
 */
import { FIELD_TYPES } from '@moderntribe/events-pro/blocks/additional-fields/utils';
import { actions, types, selectors } from '@moderntribe/events-pro/data/blocks/additional-fields';
import { string } from '@moderntribe/common/utils';

export function* setInitialState( props ) {
	const { payload = {} } = props;
	const { name = '', get = noop } = payload;
	const initialValues = get( 'initialValues', {} );

	yield all( [
		put( actions.addField( name ) ),
		put( actions.setFieldIsPristine( name, get( 'isPristine' ) ) ),
		put( actions.setFieldOutput( name, get( 'output' ) ) ),
		put( actions.setFieldType( name, initialValues.type ) ),
		put( actions.setFieldLabel( name, initialValues.label ) ),
		put( actions.setFieldOptions( name, initialValues.options ) ),
		put( actions.setFieldMetaKey( name, initialValues.metaKey ) ),
		put( actions.setFieldValue( name, get( 'value' ) ) ),
		put( actions.setFieldChange( name ) ),
	] );
}

export function* setPristineState( props ) {
	const { payload = {} } = props;
	const { name = '' } = payload;
	const value = yield select( selectors.getFieldValue, payload );
	const isPristine = value === null || '' === value || isEmpty( value );
	yield put( actions.setFieldIsPristine( name, isPristine ) );
}

export function* appendFieldValue( props ) {
	const { payload = {} } = props;
	const { name = '', value } = payload;
	const current = yield select( selectors.getFieldCheckboxValue, payload );
	const newValue = [ ...current, value ]
		.filter( identity )
		.join( '|' );
	yield put( actions.setFieldValue( name, newValue ) );
}

export function* removeFieldValue( props ) {
	const { payload = {} } = props;
	const { name = '', value } = payload;
	const current = yield select( selectors.getFieldCheckboxValue, payload );
	const newValue = current
		.filter( ( text ) => text !== value )
		.join( '|' );
	yield put( actions.setFieldValue( name, newValue ) );
}

export function* onFieldBlur( props ) {
	const { payload = {} } = props;
	const { name = '' } = payload;
	const type = yield select( selectors.getFieldType, payload );
	yield put( actions.setFieldBlurWithType( name, type ) );
}

export function* setTextFieldOutput( props ) {
	const { payload = {} } = props;
	const { name = '' } = payload;
	const value = yield select( selectors.getFieldValue, payload );
	yield put( actions.setFieldOutput( name, value ) );
}

export function* setDropdownOutput( props ) {
	const { payload = {} } = props;
	const { name = '' } = payload;
	const { label = '' } = yield select( selectors.getFieldDropdownValue, payload );
	yield put( actions.setFieldOutput( name, label ) );
}

export function* setCheckboxOutput( props ) {
	const { payload = {} } = props;
	const { name = '' } = payload;
	const { values, listDivider, dividerEnd } = yield all( {
		values: select( selectors.getFieldCheckboxValue, payload ),
		listDivider: select( selectors.getFieldDividerList, payload ),
		dividerEnd: select( selectors.getFieldDividerEnd, payload ),
	} );
	yield put( actions.setFieldOutput( name, string.wordsAsList( values, listDivider, dividerEnd ) ) );
}

export default function* watchers() {
	yield takeEvery( types.SET_ADDITIONAL_FIELD_INITIAL_STATE, setInitialState );
	yield takeEvery( types.SET_ADDITIONAL_FIELD_CHANGE, setPristineState );
	yield takeEvery( types.APPEND_ADDITIONAL_FIELD_VALUE, appendFieldValue );
	yield takeEvery( types.REMOVE_ADDITIONAL_FIELD_VALUE, removeFieldValue );
	yield takeEvery( types.SET_ADDITIONAL_FIELD_BLUR, onFieldBlur );
	yield takeEvery(
		[
			`${ types.SET_ADDITIONAL_FIELD_BLUR }/${ FIELD_TYPES.text }`,
			`${ types.SET_ADDITIONAL_FIELD_BLUR }/${ FIELD_TYPES.radio }`,
			`${ types.SET_ADDITIONAL_FIELD_BLUR }/${ FIELD_TYPES.url }`,
			`${ types.SET_ADDITIONAL_FIELD_BLUR }/${ FIELD_TYPES.textarea }`,
		],
		setTextFieldOutput,
	);
	yield takeEvery(
		`${ types.SET_ADDITIONAL_FIELD_BLUR }/${ FIELD_TYPES.dropdown }`,
		setDropdownOutput,
	);
	yield takeEvery(
		`${ types.SET_ADDITIONAL_FIELD_BLUR }/${ FIELD_TYPES.checkbox }`,
		setCheckboxOutput,
	);
}
