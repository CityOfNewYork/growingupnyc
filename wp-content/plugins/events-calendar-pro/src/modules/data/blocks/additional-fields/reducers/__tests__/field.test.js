/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events-pro/data/blocks/additional-fields';
import field, { DEFAULT_STATE } from '../field';
import { FIELD_TYPES } from '@moderntribe/events-pro/blocks/additional-fields/utils';

describe( 'Fields reducer', () => {
	let state = {};
	const fieldName = '_ecp_custom_23';

	beforeEach( () => {
		state = { allIds: [], byId: {} };
	} );

	test( 'Default reducer', () => {
		expect( field( undefined, {} ) ).toEqual( DEFAULT_STATE );
	} );

	test( 'Set ticket name', () => {
		expect( field( DEFAULT_STATE, actions.setFieldLabel( fieldName, 'Website' ) ) )
			.toMatchSnapshot();
	} );

	test( 'Set ticket value', () => {
		const action = actions.setFieldValue( fieldName, 'https://theeventscalendar.com/' );
		expect( field( DEFAULT_STATE, action ) ).toMatchSnapshot();
	} );

	test( 'Set field type', () => {
		expect( field( DEFAULT_STATE, actions.setFieldType( fieldName, FIELD_TYPES.url ) ) )
			.toMatchSnapshot();
	} );

	test( 'Set field is pristine', () => {
		expect( field( DEFAULT_STATE, actions.setFieldIsPristine( fieldName, true ) ) )
			.toMatchSnapshot();
		expect( field( DEFAULT_STATE, actions.setFieldIsPristine( fieldName, false ) ) )
			.toMatchSnapshot();
	} );

	test( 'Set field options', () => {
		const action = actions.setFieldOptions( fieldName, [ 'Pizza', 'Cheese Cake', 'Cheese' ] );
		expect( field( DEFAULT_STATE, action ) ).toMatchSnapshot();
	} );

	test( 'Set field divider list', () => {
		const action = actions.setFieldDividerList( fieldName, ', ' );
		expect( field( DEFAULT_STATE, action ) ).toMatchSnapshot();
	} );

	test( 'Set field divider end', () => {
		const action = actions.setFieldDividerEnd( fieldName, ' & ' );
		expect( field( DEFAULT_STATE, action ) ).toMatchSnapshot();
	} );

	test( 'append field value', () => {
		const action = actions.appendFieldValue( fieldName, 'Chocolate' );
		const newState = {
			...DEFAULT_STATE,
			value: [],
		};
		expect( field( newState, action ) ).toMatchSnapshot();
	} );

	test( 'remove field value', () => {
		const action = actions.removeFieldValue( fieldName, 'Chocolate' );
		const newState = {
			...DEFAULT_STATE,
			value: [ 'Chocolate' ],
		};
		expect( field( newState, action ) ).toMatchSnapshot();
	} );

	test( 'set meta key value', () => {
		const action = actions.setFieldMetaKey( fieldName, '_additional_field' );
		expect( field( DEFAULT_STATE, action ) ).toMatchSnapshot();
	} );

	test( 'field output', () => {
		const action = actions.setFieldOutput( fieldName, 'My custom output' );
		expect( field( DEFAULT_STATE, action ) ).toMatchSnapshot();
	} );
} );
