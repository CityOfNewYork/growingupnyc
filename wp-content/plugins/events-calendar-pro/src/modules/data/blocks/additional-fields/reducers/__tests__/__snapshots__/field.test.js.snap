// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields reducer Set field divider end 1`] = `
Object {
  "dividerEnd": " & ",
  "dividerList": ", ",
  "isPristine": true,
  "label": "",
  "metaKey": "",
  "options": Array [],
  "output": "",
  "type": "",
  "value": "",
}
`;

exports[`Fields reducer Set field divider list 1`] = `
Object {
  "dividerEnd": " & ",
  "dividerList": ", ",
  "isPristine": true,
  "label": "",
  "metaKey": "",
  "options": Array [],
  "output": "",
  "type": "",
  "value": "",
}
`;

exports[`Fields reducer Set field is pristine 1`] = `
Object {
  "dividerEnd": " & ",
  "dividerList": ", ",
  "isPristine": true,
  "label": "",
  "metaKey": "",
  "options": Array [],
  "output": "",
  "type": "",
  "value": "",
}
`;

exports[`Fields reducer Set field is pristine 2`] = `
Object {
  "dividerEnd": " & ",
  "dividerList": ", ",
  "isPristine": false,
  "label": "",
  "metaKey": "",
  "options": Array [],
  "output": "",
  "type": "",
  "value": "",
}
`;

exports[`Fields reducer Set field options 1`] = `
Object {
  "dividerEnd": " & ",
  "dividerList": ", ",
  "isPristine": true,
  "label": "",
  "metaKey": "",
  "options": Array [
    "Pizza",
    "Cheese Cake",
    "Cheese",
  ],
  "output": "",
  "type": "",
  "value": "",
}
`;

exports[`Fields reducer Set field type 1`] = `
Object {
  "dividerEnd": " & ",
  "dividerList": ", ",
  "isPristine": true,
  "label": "",
  "metaKey": "",
  "options": Array [],
  "output": "",
  "type": "url",
  "value": "",
}
`;

exports[`Fields reducer Set ticket name 1`] = `
Object {
  "dividerEnd": " & ",
  "dividerList": ", ",
  "isPristine": true,
  "label": "Website",
  "metaKey": "",
  "options": Array [],
  "output": "",
  "type": "",
  "value": "",
}
`;

exports[`Fields reducer Set ticket value 1`] = `
Object {
  "dividerEnd": " & ",
  "dividerList": ", ",
  "isPristine": true,
  "label": "",
  "metaKey": "",
  "options": Array [],
  "output": "",
  "type": "",
  "value": "https://theeventscalendar.com/",
}
`;

exports[`Fields reducer append field value 1`] = `
Object {
  "dividerEnd": " & ",
  "dividerList": ", ",
  "isPristine": true,
  "label": "",
  "metaKey": "",
  "options": Array [],
  "output": "",
  "type": "",
  "value": Array [],
}
`;

exports[`Fields reducer field output 1`] = `
Object {
  "dividerEnd": " & ",
  "dividerList": ", ",
  "isPristine": true,
  "label": "",
  "metaKey": "",
  "options": Array [],
  "output": "My custom output",
  "type": "",
  "value": "",
}
`;

exports[`Fields reducer remove field value 1`] = `
Object {
  "dividerEnd": " & ",
  "dividerList": ", ",
  "isPristine": true,
  "label": "",
  "metaKey": "",
  "options": Array [],
  "output": "",
  "type": "",
  "value": Array [
    "Chocolate",
  ],
}
`;

exports[`Fields reducer set meta key value 1`] = `
Object {
  "dividerEnd": " & ",
  "dividerList": ", ",
  "isPristine": true,
  "label": "",
  "metaKey": "_additional_field",
  "options": Array [],
  "output": "",
  "type": "",
  "value": "",
}
`;
